subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/android/AndroidPublishPlugin.java

```diff
@@ -0,0 +1,82 @@
+import org.shipkit.gradle.configuration.AndroidPublishConfiguration;
+import org.shipkit.gradle.configuration.ShipkitConfiguration;
+import org.shipkit.internal.gradle.configuration.ShipkitConfigurationPlugin;
+import org.shipkit.internal.gradle.snapshot.LocalSnapshotPlugin;
+import org.shipkit.internal.gradle.util.GradleDSLHelper;
+import org.shipkit.internal.gradle.util.PomCustomizer;
+
+import static org.shipkit.internal.gradle.java.JavaPublishPlugin.MAVEN_LOCAL_TASK;
+import static org.shipkit.internal.gradle.java.JavaPublishPlugin.PUBLICATION_NAME;
+
+/**
+ * Publishing Android libraries using 'maven-publish' plugin.
+ * Intended to be applied in individual Android library submodule.
+ * Applies following plugins and tasks and configures them:
+ *
+ * <ul>
+ *     <li>maven-publish</li>
+ * </ul>
+ *
+ * Other features:
+ * <ul>
+ *     <li>Configures Gradle's publications to publish Android library</li>
+ *     <li>Configures 'build' task to depend on 'publishJavaLibraryToMavenLocal'
+ *          to flesh out publication issues during the build</li>
+ *     <li>Configures 'snapshot' task to depend on 'publishJavaLibraryToMavenLocal'</li>
+ * </ul>
+ */
+public class AndroidPublishPlugin implements Plugin<Project> {
+
+    private final static Logger LOG = Logging.getLogger(AndroidPublishPlugin.class);
+    private final static String ANDROID_PUBLISH_EXTENSION = "androidPublish";
+
+    public void apply(final Project project) {
+        final AndroidPublishConfiguration androidPublishConfiguration = project.getExtensions().create(ANDROID_PUBLISH_EXTENSION, AndroidPublishConfiguration.class);
+
+        final ShipkitConfiguration conf = project.getPlugins().apply(ShipkitConfigurationPlugin.class).getConfiguration();
+
+        project.getPlugins().apply(LocalSnapshotPlugin.class);
+        Task snapshotTask = project.getTasks().getByName(LocalSnapshotPlugin.SNAPSHOT_TASK);
+        snapshotTask.dependsOn(MAVEN_LOCAL_TASK);
+
+        project.getPlugins().apply("maven-publish");
+
+        BintrayExtension bintray = project.getExtensions().getByType(BintrayExtension.class);
+        bintray.setPublications(PUBLICATION_NAME);
+
+        project.getPlugins().withId("com.android.library", plugin -> {
+            project.afterEvaluate(evaluatedProject -> {
+
+                GradleDSLHelper.publications(project, publications -> {
+                    MavenPublication p = publications.create(PUBLICATION_NAME, MavenPublication.class, publication -> {
+                        publication.setArtifactId(androidPublishConfiguration.getArtifactId());
+
+                        SoftwareComponent releaseComponent = project.getComponents().findByName("release");
+                        if (releaseComponent == null) {
+                            throw new GradleException("'release' component not found in project. " +
+                                "Make sure you are using Android Gradle Plugin 3.6.0-beta05 or newer.");
```

Comment:
This message may be confusing. This PR include com.android.tools.build:gradle:3.6.0-beta05 so it should be applied by Shipkit autmatically. Or documentation should describe that the user needs to apply it manually.

Reply:
User has to apply `com.android.library` plugin to every Android Library plugin.

However indeed, Gradle dependency on `com.android.tools.build:gradle` is not needed since we are not using any classes from it. We only refer to plugin and component by names given in plain strings. I'll remove the dependency.

I'll also mention min AGP version in documentation.